src/olympia/devhub/tests/test_forms.py

```diff
@@ -37,6 +38,23 @@ def test_firefox_default_selected(self):
         form = forms.NewUploadForm(data, request=request)
         assert form.fields['compatible_apps'].initial == [amo.FIREFOX.id]
 
+    def test_previous_compatible_apps_initially_selected(self):
+        addon = addon_factory()
+
+        appversion = AppVersion.objects.create(
+            application=amo.ANDROID.id, version='1.0')
+        ApplicationsVersions.objects.create(
+            version=addon.current_version, application=amo.ANDROID.id,
+            min=appversion, max=appversion)
+
+        upload = FileUpload.objects.create(valid=False)
+        data = {'upload': upload.uuid}
+        request = req_factory_factory('/', post=True, data=data)
+        request.user = user_factory()
+        form = forms.NewUploadForm(data, request=request, addon=addon)
+        assert form.fields['compatible_apps'].initial == [
+            amo.FIREFOX.id, amo.ANDROID.id]
```

Comment:
can we test first that `form.fields['compatible_apps'].initial` would otherwise not contain `amo.ANDROID.id` if `addon.current_version` hadn't been changed?